src/pages/products/Products.jsx

```diff
@@ -30,26 +30,27 @@ const Products = () => {
 
   const addToCartHandler = async (product) => {
     try {
+    
       if (!currentUser) {
         navigate("/login");
       } else {
-        let cart = await getCart();
-        let item = cart.data.cart.find(
-          (cartProduct) => cartProduct._id === product._id
-        );
-        if (item) {
-          navigate("/cart");
-        } else {
-          let status = await addToCart(product);
-          if (status === 201) {
-            navigate("/cart");
-          }
-        }
+        let result = isProductExist(product._id);
+        
+        if(result==="Go to cart"){
+            navigate("/cart")
+            return;
+        } 
+        await addToCart(product);
       }
     } catch (error) {
       console.log(error);
     }
   };
+  const isProductExist=(id)=>{
+      let isExist = cartState.cart.find(item=> item._id === id);
```

Comment:
`find` would also work here but `includes` would be better since we are just checking whether that item is in cart or not and want to just have true or false value, we don't need that item, so `includes` would be better.

Reply:
Sure Atul